transportable-udfs-avro/src/main/java/com/linkedin/transport/avro/AvroWrapper.java

```diff
@@ -43,27 +43,69 @@ private AvroWrapper() {
 
   public static StdData createStdData(Object avroData, Schema avroSchema) {
     switch (avroSchema.getType()) {
-      case INT:
+      case INT: {
+        if (!(avroData instanceof Integer)) {
+          throw new IllegalArgumentException("Unsupported type for Avro integer: " + avroData.getClass());
+        }
         return new AvroInteger((Integer) avroData);
-      case LONG:
+      }
+      case LONG: {
+        if (!(avroData instanceof Long)) {
+          throw new IllegalArgumentException("Unsupported type for Avro long: " + avroData.getClass());
+        }
```

Comment:
Could you please explain why those checks are required? How can we end up here?

Reply:
Just for better error message rather than relying on Java class cast exception. When we use AvroWrapper outside of Transport UDF, this may happen.

Reply:
Thanks! Could you please show an example call? This method is mostly internal to the framework. Do you expect the user to call it directly, or still will it be called from the framework? To see what I am referring to, see usages in other engines (Preto, Spark, etc).

Reply:
As I mentioned before, we need to use it for Limestone portable engine. The engine will need to create StdData from the actual Avro record (reading from Kafka or HDFS, for example).

Reply:
I would recommend removing that check to keep the parallelism of implementation in different engines, since we do not have such a check in others. To keep parallelism, we either remove it from here or add it to others. Reason for not adding it to others is that it will affect performance (while still throwing an error in both cases), and it will not be very helpful since all the createStdData() calls are made from within Transport and they exactly know what type it expects (so it is not up to the user to make mistakes). I think the case of Limestone is similar, where it is an engine that knows the contract, and there is no need to check whether Limestone is doing the right call or not. Such checks are better covered with unit tests.